llvm/docs/GitHub.rst

```diff
@@ -46,11 +73,106 @@ The default commit message for a squashed pull request is the pull request
 description, so this will allow reviewers to review the commit message before
 approving the commit.
 
+When pushing to your branch, make sure you push to the correct fork. Check your
+remotes with:
+
+::
+
+  git remote -v
+
+And make sure you push to the remote that's pointing to your fork.
+
+Rebasing Pull Requests and Force Pushes
+---------------------------------------
+In general, you should avoid rebasing a Pull Request and force pushing to the
+branch that's the root of the Pull Request during the review. This action will
+make the context of the old changes and comments harder to find and read.
+
+Sometimes, a rebase might be needed to update your branch with a fix for a test
+or in some dependent code.
+
+After your PR is reviewed and accepted, you want to rebase your branch to ensure
+you won't encounter merge conflicts when landing the PR.
+
+Landing your change
+-------------------
+When your PR has been accepted you can use the web interface to land your change.
+The button that should be used is called `Squash and merge` and after you can
+select the option `Delete branch` to delete the branch from your fork.
```

Comment:
What happens to the review history when the branch is removed? Is it still possible to go back and follow up on all the review comments five years later if the branch is removed, or is all that stored somewhere else.

And if it is stored somewhere else, then I don't understand how rebases of the branch in my fork would "make the context of the old changes and comments harder to find and read". Or is that always the case anyhow when doing the final "squash and merge". In Phabricator I think comment line up nicely if for example looking at a specific diff from the patches history (i.e. a diff matching with the timeline when the comment was written).

Reply:
The review is always available on github.

> I don't understand how rebases of the branch in my fork would "make the context of the old changes and comments harder to find and read".

I should have said, the "last state of the review" is always available on github. When we force-push (which is pretty much required after a rebase), previous states are gone forever. Github only knows how to track your PR against *one* base; if we rebase and things change, some comments may be dangling

Reply:
Ok, thanks for explaining. But then the --delete part should be ok!